sale_order_product_recommendation/wizards/sale_order_recommendation.py

```diff
@@ -128,8 +132,9 @@ def _generate_recommendations(self):
         found_dict = {l["product_id"][0]: l for l in found_lines}
         recommendation_lines = self.env["sale.order.recommendation.line"]
         existing_product_ids = set()
-        # Always recommend all products already present in the linked SO
-        for line in self.order_id.order_line:
+        # Always recommend all products already present in the linked SO except delivery
+        # carrier products
+        for line in self.order_id.order_line.filtered(lambda ln: not ln.is_delivery):
```

Comment:
```suggestion
        lines = self.order_id.order_line
        if "is_delivery" in self.env["sale.order.line"]._fields:
            lines = lines.filtered(lambda ln: not ln.is_delivery)
        for line in lines:
```

Reply:
This one is missing

Reply:
Here we can use method _is_delivery() from sale module